src/app/core/component/graph-display/graph-display.component.ts

```diff
@@ -214,9 +256,29 @@ export class GraphDisplayComponent implements AfterViewInit, OnChanges {
             this.patternLanguage = this.patternGraphData.patternLanguage;
             this.patternView = this.patternGraphData.patternView;
             this.nodes = GraphDisplayComponent.mapPatternsToNodes(this.patterns);
+            if (this.patternView) {
+                this.patternLanguages = this.getPatterns(this.patternGraphData.patternLanguages);
+            }
         }
     }
 
+    private getPatterns(patternLanguages: Array<PatternLanguage>) {
+        const amountOfPLs = patternLanguages.length;
```

Comment:
I'd prefer to load the patterns of a language in case when you need them (in the cards view modal for adding patterns, we trigger loading of the patterns when a pattern language is expanded - this could save us a ton of requests :D )